### PR TITLE
Added decorator to measure renderToStringAsync execution time

### PR DESCRIPTION
**Pull request by wredzio**

### CODE REVIEW

File: React/Extensions/ReactExtensions.cs [hunk 1]
```diff
@@ -20,7 +21,10 @@ public static class ReactExtensions
     {
         services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
         ReactServiceCollectionExtensions.AddReact(services);
```
Comment (pawil31) [inline React/Extensions/ReactExtensions.cs:28]: We should not add logging decorator as default behavior. There should be an option to project who is using this package to decide if logging is needed.

File: React/Services/IReactService.cs [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
+    
+    public ReactServiceRenderToStringAsyncExecutionTimeDecorator(IReactService service, ILogger<ReactServiceRenderToStringAsyncExecutionTimeDecorator> logger)
+    {
+        _service = service;
+        _logger = logger;
+    }
+
+    public async Task<IHtmlContent> RenderToStringAsync<T>(IHtmlHelper htmlHelper, string componentName, T props)
+    {
+        var watch = System.Diagnostics.Stopwatch.StartNew();
```
Comment (pawil31) [inline React/Services/IReactService.cs]: remove namepsace, add to usings

File: React/Services/IReactService.cs [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
+        _logger = logger;
+    }
+
+    public async Task<IHtmlContent> RenderToStringAsync<T>(IHtmlHelper htmlHelper, string componentName, T props)
+    {
+        var watch = System.Diagnostics.Stopwatch.StartNew();    
+        var renderedString = await _service.RenderToStringAsync(htmlHelper, componentName, props);
+        var elapsedMs = watch.ElapsedMilliseconds;
+
+        _logger.Log(LogLevel.Debug, $"RenderToStringAsync for component {componentName} lasted: {elapsedMs}");
```
Comment (pawil31) [inline React/Services/IReactService.cs]: This is no how we should do logging. Do not use string interpolation. We should use method like this:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.log?view=dotnet-plat-ext-7.0#microsoft-extensions-logging-loggerextensions-log(microsoft-extensions-logging-ilogger-microsoft-extensions-logging-loglevel-system-string-system-object())
and give a names to fields. This way we can use them in logs. When you let Logging concatenate your string you can use name in logs/App insights and query against that ( I mean you can query against property in logs componentName or elapsedMs. If you do string interpolation it just appears as string,. 
`logger.Log(LogLevel.Debug, "RenderToStringAsync for component {componentName} lasted: {elapsedMs}",componentName,elapsedMs);`

File: React/Services/IReactService.cs [hunk 4]
```diff
@@ -1,11 +1,34 @@
...
 
-internal interface IReactService
+internal class ReactServiceRenderToStringAsyncExecutionTimeDecorator : IReactService
 {
-    Task<IHtmlContent> RenderToStringAsync<T>(IHtmlHelper htmlHelper, string componentName, T props);
-    IHtmlContent InitJavascript(IHtmlHelper htmlHelper);
+    private readonly IReactService _service;
+    private readonly ILogger<ReactServiceRenderToStringAsyncExecutionTimeDecorator> _logger;
+    
+    public ReactServiceRenderToStringAsyncExecutionTimeDecorator(IReactService service, ILogger<ReactServiceRenderToStringAsyncExecutionTimeDecorator> logger)
```
Comment (konrad-galczynski) [inline React/Services/IReactService.cs]: please consider renaming to pattern `addedBehavior DecoratedThing Decorator` so in this case: `ExecutionTimeReactServiceRenderToStringAsyncDecorator` as this vastly increase readability

File: React/Extensions/ReactExtensions.cs [hunk 5]
```diff
@@ -10,27 +10,66 @@
 using Jering.Javascript.NodeJS;
 using React.Services;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 
 namespace React.Extensions;
 
+public class ReactOptions
+{
+    public bool EnableExecutionTimeReactServiceRenderToStringAsync { get; set; } = false;
```
Comment (pawil31) [inline React/Extensions/ReactExtensions.cs]: There is nothing about logging execution time in property name, and this is what a decorator is doing.

**Review by konrad-galczynski — APPROVED** [review]

**Pull request merged**